client/question/services/EventHandlerService.js

```diff
@@ -19,7 +19,38 @@
 tie.factory('EventHandlerService', [
   '$http', 'ServerHandlerService', function($http, ServerHandlerService) {
 
+
+    /**
+     * Global object to keep track of the current batch of events to send.
+     * @type [Event]
+     */
+    var currentEventBatch = [];
+
+    /**
+     * Submits the current batch of event data to TIE's backend.
+     *
+     */
+    var sendCurrentEventBatch = function() {
+      var data = {
+        events: currentEventBatch,
+        createdMsec: (new Date()).getTime()
+      };
+      return $http.post('/ajax/event/send_event_batch', data).then(
+        function() {
+          while (currentEventBatch.length > 0) {
+            currentEventBatch.pop();
+          }
+        }, function() {
+          // Otherwise, we should do nothing.
+          // Since the call errored, don't dump the current batch.
+      });
+    };
+
     return {
+      currentEventBatch: currentEventBatch,
```

Comment:
Mm, I think you want currentEventBatch to be private. Expose it via a getter if needed for tests. You can also signal that it's private by prepending an underscore, e.g. https://github.com/google/tie/blob/master/client/question/services/FeedbackGeneratorService.js#L710

Reply:
okay.